### ISSUE TITLE
Stokhos:  Using non-supported RUN_SERIAL with tribits_add_executable_and_test()

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#11411]
## Bug Report

@trilinos/stokhos 

### Description

It would appear that there are some Stokhos tests that are defined using [`tribits_add_executable_and_test()`](https://tribitspub.github.io/TriBITS/users_guide/index.html#tribits-add-executable-and-test) that are passing in `RUN_SERIAL`.  But it appears that  `tribits_add_executable_and_test()` does not (yet) support `RUN_SERIAL` and the configure of Trilinos shows CMake warnings like:

```
CMake Warning at cmake/tribits/core/utils/MessageWrapper.cmake:77 (message):
  Arguments passed in unrecognized.  PARSE_UNPARSED_ARGUMENTS = 'RUN_SERIAL'
Call Stack (most recent call first):
  cmake/tribits/core/utils/TribitsParseArgumentsHelpers.cmake:76 (message_wrapper)
  cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:167 (tribits_check_for_unparsed_arguments)
  packages/stokhos/test/UnitTest/CMakeLists.txt:557 (TRIBITS_ADD_EXECUTABLE_AND_TEST)
```

For a fuller set of warnings, see the build:

* [`PR-11397-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-1113`](https://trilinos-cdash.sandia.gov/build/663321/configure)

### Steps to Reproduce

See the Trilinos PR builds.

**Comment by bartlettroscoe** [comment]
This waning was likely not seen before because the call to `tribits_check_for_unparsed_arguments()` was added later.

The correct solution is to add support for 	`RUN_SERIAL` to `tribits_add_executable_and_test()`.

**Comment by bartlettroscoe** [comment]
FYI: This is fixed in https://github.com/TriBITSPub/TriBITS/pull/555.  I will merge that to TriBITS 'master' and then add that to the TriBITS snapshot as part of PR #11380.

**Comment by etphipp** [comment]
Thanks @bartlettroscoe!

**Comment by bartlettroscoe** [comment]
This should be resolved with the merge of PR #11380.  And you can see these warnings gone in PR build:

* [`PR-11380-test-rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables-1117`](https://trilinos-cdash.sandia.gov/build/664547/configure)

with the only warnings being:

```
CMake Warning at packages/kokkos-kernels/cmake/kokkoskernels_eti_devices.cmake:79 (MESSAGE):
  The CMake option CMAKE_CXX_USE_RESPONSE_FILE_FOR_OBJECTS is either
  undefined or OFF.  Please set
  CMAKE_CXX_USE_RESPONSE_FILE_FOR_OBJECTS:BOOL=ON when building with CUDA and
  complex double enabled.
Call Stack (most recent call first):
  packages/kokkos-kernels/CMakeLists.txt:176 (INCLUDE)
```

and

```
CMake Deprecation Warning at packages/sacado/test/GTestSuite/googletest/CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

and

```
CMake Deprecation Warning at packages/sacado/test/GTestSuite/googletest/googletest/CMakeLists.txt:56 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

and

```
CMake Warning at packages/ifpack/CMakeLists.txt:90 (MESSAGE):
  Ifpack_SupportGraph requires Boost, but CUDA is enabled.  The parts of
  Boost that it uses do not currently build correctly with CUDA.
  Ifpack_SupportGraph users are unlikely to depend on CUDA.  Thus, we are
  turning off Ifpack_SupportGraph support.
```

Closing this as complete.
